File: Movie/Movie.Web/Modules/ServerTypes/MovieDB/PersonColumns.ts
﻿import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { Gender } from "../Modules/MovieDB.Person.Gender";
import { PersonRow } from "./PersonRow";

export interface PersonColumns {
    PersonId: Column<PersonRow>;
    FirstName: Column<PersonRow>;
    LastName: Column<PersonRow>;
    FullName: Column<PersonRow>;
    BirthDate: Column<PersonRow>;
    BirthPlace: Column<PersonRow>;
    Gender: Column<PersonRow>;
    Height: Column<PersonRow>;
}

export class PersonColumns extends ColumnsBase<PersonRow> {
    static readonly columnsKey = 'MovieDB.Person';
    static readonly Fields = fieldsProxy<PersonColumns>();
}

queueMicrotask(() => [Gender]); // referenced types

File: Movie/Movie.Web/Modules/MovieDB/Person/PersonColumns.cs
using Movie.Modules.MovieDB.Person;
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace Movie.MovieDB.Columns;

[ColumnsScript("MovieDB.Person")]
[BasedOnRow(typeof(PersonRow), CheckNames = true)]
public class PersonColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int PersonId { get; set; }
    [EditLink]
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string FullName { get; set; }
    public DateTime BirthDate { get; set; }
    public string BirthPlace { get; set; }
    public Gender Gender { get; set; }
    public int Height { get; set; }
}